utils/git.go

```diff
@@ -201,6 +260,26 @@ func (gm *GitManager) CreateBranchAndCheckout(branchName string, keepLocalChange
 	return err
 }
 
+func (gm *GitManager) createBranchAndCheckoutToHash(hash string, keepLocalChanges bool) error {
+	var checkoutConfig *git.CheckoutOptions
+	if keepLocalChanges {
```

Comment:
In the single usage of this new function you pass keepLocalChanges=false, so why do we need to even refer to that in this function?
Is this intended for future usage or plans? If not we can consider deleting it